### ISSUE TITLE
camelCase + capitalized abbreviation prop name: works in 1.0.26, breaks in 1.0.27+

### ISSUE THREAD

**Issue by CrookedNumber** [vuejs/vue#3850]
### Vue.js version

1.0.27 + 
### Reproduction Link

https://jsfiddle.net/zb7872qz/
### Steps to reproduce

< 1.0.27, you can pass down a prop with a name like `sectionID` by naming it `section-id` in the parent component. I have no idea if this is the intended consequence or the suggested convention/practice, but it does work. Example: https://jsfiddle.net/g1srmert/

But => 1.0.27, this will break. Example: https://jsfiddle.net/zb7872qz/

After finding this ticket https://github.com/vuejs/vue/issues/3187, I guessed that a fix would involve passing a prop with the name `section-i-d` -- which totally makes sense in retrospect (how would the parser know you meant sectionID and not sectionId?). Solved: https://jsfiddle.net/yz4hw2ds/

Note that the above fix (`section-i-d`) will _break_ code in Vue <1.0.26 Example: https://jsfiddle.net/75tntoqt/

In other words, this is definitely an edge case. That being said, it _does_ break existing code that _did_ work (for better or worse) in a prior patch version.
### What is Expected?

Continuity of passing down props with names like `sectionID`.  To reiterate: I have no idea if this is the intended consequence or the suggested convention/practice, but it does work.
### What is actually happening?

I don't know if this is a tiny manifestation of a smaller bug or even worthy of a bug fix (assuming that's even possible). But it does seem like there should be a warning (somewhere?) and/or slightly more explicit docs here https://vuejs.org/guide/components.html#camelCase-vs-kebab-case and here https://vuejs.org/2016/02/06/common-gotchas/

**Comment by devillom** [comment]
+1

**Comment by l0rn** [comment]
We are using Vue 2.3.4 and just realized the following:
If you use camelCase in the HTML prop it will just work fine, although the documentation suggests kebab-case for those. No problem here BUT if I use the webpack UglifyPlugin to minify the bundle all those camelCase props will lead to errors. This bug is pretty nasty and I think camelCase should be forbidden in general for html-props

**Comment by yyx990803** [comment]
Closing 1.x issues as 1.x is now end of life and will only receive critical security patches.
